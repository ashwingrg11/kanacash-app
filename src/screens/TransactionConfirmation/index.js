import React from 'react';
import {View, StyleSheet} from 'react-native';
import GenericView from '~/components/ui/GenericView';
import Header from '~/components/ui/Header';
import {useSelector, useDispatch} from 'react-redux';
import {setError} from '~/store/actions/Error';
import * as api from '~/services/axios/Api';
import {NAVIGATION_TO_TRANSACTION_SUCCESS_SCREEN} from '../../navigation/routes';
import {BoldText} from '~/components/ui/Text';
import FooterButton from '~/components/ui/FooterButton';
import theme from '~/components/theme/Style';
import {AUTHORIZATION} from '~/constants/info';
import {ScrollView} from 'react-native-gesture-handler';
import {hideLoader, showLoader} from '~/store/actions/LoaderAction';

export default function TransactionConfirmation({navigation}) {
  const dispatch = useDispatch();
  const transaction = useSelector(state => state.transaction);

  const onPressContinue = () => {
    const {
      feeAmount,
      exchangeRate,
      senderAmount,
      recipientAmount,
      recipientCurrency,
      destinationCountry,
      recipientId,
      payoutMethod,
      fundingSource,
      senderFundingAccountId,
      recipientBankId,
      payerId,
    } = transaction;

    const body = {
      feeAmount,
      exchangeRate,
      senderAmount,
      recipientAmount,
      recipientCurrency,
      destinationCountry,
      recipientId,
      payoutMethod,
      fundingSource,
      senderFundingAccountId,
      recipientAccountId: recipientBankId,
      payerId,
      remittancePurpose: 'other',
    };
    console.log('body', body);
    dispatch(showLoader());
    api
      .createTransaction(body)
      .then(res => {
        console.log('response', res);
        if (res.status === 200) {
          dispatch(hideLoader());
          navigation.navigate(NAVIGATION_TO_TRANSACTION_SUCCESS_SCREEN);
        } else {
          let modalConfig = {
            message: res.data.message,
            message_title: 'Transaction Failed',
          };
          dispatch(hideLoader());
          dispatch(setError(modalConfig));
        }
      })
      // eslint-disable-next-line handle-callback-err
      .catch(err => {
        dispatch(hideLoader());
        let modalConfig = {
          message: err?.data?.message
            ? JSON.parse(err.data.message)?.message
            : 'Error while updating receiver',
          message_title: 'Transaction Failed',
        };
        dispatch(setError(modalConfig));
      });
  };

  return (
    <GenericView padding header={<Header title="Confirmation" />}>
      <ScrollView
        showsVerticalScrollIndicator={false}
        contentContainerStyle={styles.contentContainerStyle}>
        <View style={styles.containerStyle}>
          <BoldText text={'Authorization'} style={styles.titleText} />
          <AUTHORIZATION />
        </View>
        <FooterButton text="Yes, Continue" onPress={onPressContinue} />
      </ScrollView>
    </GenericView>
  );
}

const styles = StyleSheet.create({
  contentContainerStyle: {flexGrow: 1, justifyContent: 'center'},
  containerStyle: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  titleText: {
    marginVertical: 10,
    color: theme.primaryColor,
    textAlign: 'center',
  },
  termContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 20,
  },
  descriptionText: {
    fontFamily: theme.themeFontRegular,
    fontSize: theme.fontSizeRegular,
    lineHeight: 25,
    color: '#5C5C5C',
    textAlign: 'center',
  },
  boldText: {
    fontWeight: '700',
    color: '#1f1f1f',
  },
});
