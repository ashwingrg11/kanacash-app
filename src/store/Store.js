import {applyMiddleware, createStore} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';
import {composeWithDevTools} from 'redux-devtools-extension';
import AsyncStorage from '@react-native-community/async-storage';

const config = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: [],
  blackList: [],
};

const reducer = persistReducer(config, rootReducer);
const middleware = composeWithDevTools(applyMiddleware(thunk));

export default data => {
  const store = createStore(reducer, middleware);
  const persistor = persistStore(store);
  return {persistor, store};
};
