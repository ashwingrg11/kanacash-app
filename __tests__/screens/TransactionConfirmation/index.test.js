import 'react-native';
import React from 'react';

import {SafeAreaProvider} from 'react-native-safe-area-context';

import TransactionConfirmation from '~/screens/TransactionConfirmation';
import Store from '~/store/Store';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('< TransactionConfirmation/>', () => {
  it('TransactionConfirmation renders correctly', () => {
    const tree = renderer
      .create(
        <SafeAreaProvider store={Store}>
          <TransactionConfirmation />,
        </SafeAreaProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
