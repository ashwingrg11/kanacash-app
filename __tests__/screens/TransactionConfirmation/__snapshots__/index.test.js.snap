// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< TransactionConfirmation/> TransactionConfirmation renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
